geiger/src/graph/Graph.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Collection from '../util/Collection';
+import Events from '../util/Events';
+import NodeWithPort from './NodeWithPort';
+import Edge, {generateEdgePortIdsByMessage} from './Edge';
+import {EdgeTypes} from './EdgeTypes';
+
+// /<reference path="../jsdoc.types.js" />
+
+/**
+ * A module that tracks all the nodes and edges.
+ *
+ * @class Graph
+ * @extends Events
+ */
+export default class Graph extends Events {
+  /**
+   * @constructor
+   * @param {!string} contextId
+   */
+  constructor(contextId) {
+    super();
+
+    this.contextId = contextId;
+
+    this._nodes = new Collection(null, {trackExit: true});
```

Comment:
Let's annotate the types of `_nodes` and `_edges`.

Reply:
Done.